src/utils/pgwire/src/pg_extended.rs

```diff
@@ -174,6 +190,55 @@ impl PgPortal {
     pub fn query_string(&self) -> Bytes {
         self.query_string.clone()
     }
+
+    pub async fn execute<SM: SessionManager>(
+        &mut self,
+        session: Arc<SM::Session>,
+        row_limit: usize,
+    ) -> Result<PgResponse, BoxedError> {
+        if self.result_cache.is_none() {
+            let process_res = session
+                .run_statement(cstr_to_str(&self.query_string).unwrap())
+                .await;
+
+            // Return result directly if
+            // - it's not a query result.
+            // - query result needn't cache. (row_limit == 0).
+            if !(process_res.is_ok() && process_res.as_ref().unwrap().is_query()) || row_limit == 0
+            {
+                return process_res;
+            }
+
+            // Return result need to cache.
+            self.stmt_type = Some(process_res.as_ref().unwrap().get_stmt_type());
+            self.result_cache = Some(process_res.unwrap().values().into_iter());
```

Comment:
Actually I'm thinking of reusing result_cache. Means we always cache the IntoIter of Row, only different on the scan count. For row_limit, we iter fixed times. While for simple query or without row_limit, we iter until None (or number of rows returned from execute results).

Reply:
> Actually I'm thinking of reusing result_cache. Means we always cache the IntoIter of Row, only different on the scan count. For row_limit, we iter fixed times. While for simple query or without row_limit, we iter until None (or number of rows returned from execute results).

Because the result cache is in the portal, if we want to use the result cache in the simple query, I think it will be more complicated? 
And for now, when I accept a query without row_limit(row_limit == 0 ) in extended query mode, I just return the result and do not use the result cache. I consider this way a fast path. Because when use the result cache, we should get the result, and change it to Intolter and iter it.
So in this case (without row_limit), compared with using result_cache, directly returning the result will be more efficient and simple?

Reply:
Hmmm I think change it to into iter and iter it can be efficient by simply move.

Reply:
Anyway it's just a suggestion.

Reply:
Indeed a little strange.